### ISSUE TITLE
Train scaled-up Qwen3 (1B + 4B) on zoonomia-v1-v1 for 1 epoch with transferred AdamH hparams

### ISSUE THREAD

**Issue by gonzalobenegas** [Open-Athena/marin-dna#166]
## Description

Follow-up to #160 (zoonomia projection pipeline sanity-check at 0.6B Qwen3 / 10K steps × 4096 batch on v5p-8). Now that the projection pipeline is validated and the AdamH hyperparameter-transfer heuristic from #109 is calibrated to anchor on the v0.6 DNA reference sweep (and is the basis for both [`exp135_bolinas_mix_sweep.py`](https://github.com/marin-community/marin/blob/eac/dna-bolinas-mix-sweep/experiments/dna/exp135_bolinas_mix_sweep.py) at 1B and #109's `p4B` parameter scaling run), scale up the zoonomia run: train **1B** and **4B** Qwen3 gLMs on the whole-genome cross-mammal projection dataset for one full epoch.

This is a **single-mixture** experiment: train directly on `bolinas-dna/zoonomia-v1-v1` — no per-region weighting (cf. the parallel CDS/upstream/downstream mix sweep in [`exp135_bolinas_mix_sweep.py`](https://github.com/marin-community/marin/blob/eac/dna-bolinas-mix-sweep/experiments/dna/exp135_bolinas_mix_sweep.py)).

Both runs target the same hardware (`v6e-4` / `europe-west4-a`, matching #109's full `p46M…p4B` parameter scaling sweep) so 1B and 4B are directly comparable on wall-clock, throughput, and sample efficiency. `p4B` is the upper bound of the AdamH-calibrated heuristic and is the largest size known to fit on a single TPU node in this codebase.

## Hypothesis or Goal

- Establish a VEP / LL gap baseline at 1B and 4B on `zoonomia-v1-v1` cross-mammal projection data, comparable against the 0.6B baseline in #160 and the animals-CDS / upstream / downstream baselines in #109.
- Verify both 1B and 4B train stably (no divergence, no NaN/inf, monotone train loss) end-to-end on a single TPU node for the full one-epoch budget — **57.3B tokens = 223,880,280 sequences × 256 bp/seq**, ~22.9× beyond the small-scale token horizon (T0 = 2.5×10⁹) that anchors the AdamH heuristic.

### Links

* WandB Report: (TBD — group `dna-bolinas-exp<this_issue_num>`)
* Data Browser: <https://huggingface.co/datasets/bolinas-dna/zoonomia-v1-v1>
* Reference exp135 (mix sweep): <https://github.com/marin-community/marin/blob/eac/dna-bolinas-mix-sweep/experiments/dna/exp135_bolinas_mix_sweep.py>
* Reference exp109 (parameter scaling sweep): #109
* Predecessor (0.6B sanity check): #160

### Setup

- **Dataset**: [`bolinas-dna/zoonomia-v1-v1`](https://huggingface.co/datasets/bolinas-dna/zoonomia-v1-v1) — 108-species cross-mammal projection (whole-genome), reverse-complement augmented (`add_rc: true`).
- **Train sample count**: **223,880,280 rows** (counted directly from the 64 `data/train/shard_NNNN.jsonl.zst` shards via `zstd -dc | wc -l`; per-shard min/max = 3,498,129 / 3,498,130). Matches the "~224M post-RC" claim in #149 commit `1c0b754`. HF's `datasets-server` reports `estimated_num_rows = 174,621,287` (off by ~22 %) and `num_rows = 14,082,488` from a partial 10/64-shard parquet conversion — both ignored, per the "don't trust HF estimation" instinct that motivated this verification.
- **Tokenizer**: `bolinas-dna/tokenizer-char-bos`.
- **Sequence length**: 256 (255 bp + BOS), matching the AdamH-calibrated reference regime.
- **Training format**: `DNALmDatasetFormat(lowercase_weight=0.01)` (matches #160 and exp135 — repeats downweighted in training, not in validation).
- **Single mixture**: train directly on `zoonomia-v1-v1`; no per-region weighting.

### Model configurations

Both runs on the same hardware (`v6e-4` / `europe-west4-a`, matching #109's full `p46M…p4B` parameter scaling sweep) so timing and throughput are directly comparable across sizes:

| Run | Hidden | ~Params | TPU |
|-----|--------|---------|-----|
| 1B  | 1920   | ~1.12B  | `v6e-4` |
| 4B  | per `_build_model_configs` | ~4B | `v6e-4` |

Architecture: `Qwen3Config` with `Llama3RotaryEmbeddingsConfig` RoPE — same Qwen3 family as #160. Intermediate dim, n_heads (= n_kv_heads), and n_layers are determined by [`CompletedAdamHHeuristic._build_model_config`](https://github.com/marin-community/marin/blob/a243fe5628532f8213860b1668c0259275a37774/experiments/scaling_law_sweeps/completed_adamh.py#L227). `initializer_range = 0.02` overrides the heuristic's default. (The function's return-type annotation is `LlamaConfig` but the body returns `Qwen3Config(...)` — stale annotation in marin.)

### Hyperparameters (transferred via AdamH heuristic)

v0.6 DNA reference anchor (rank 1/183 of `dna-bolinas-reference-sweep-v0.6`, eval/loss=1.228545):

| Parameter | Value |
|-----------|-------|
| `lr` | 0.0156 |
| `adam_lr` | 0.0299 |
| `beta1` | 0.6676 |
| `beta2` | 0.9067 |
| `epsilon` | 1e-15 |
| `max_grad_norm` | 0.9952 |
| `z_loss_weight` | 4.31e-06 |
| `initializer_range` | 0.02 |
| Reference batch size B0 | 16384 |
| Reference tokens T0 | 2.5e9 |

The reference is anchored on the *small-scale* "Step 1 reference sweep" of #109 (hidden=512 / ~60M params), not at the 1B/4B scale we're targeting. Per-run hparams (`lr`, `adam_lr`, `epsilon`, `beta2`) are scaled from this anchor by:

```
lr      = lr0 · √(B/B0) · (T0/T)^0.3                          (capped at max_lr = 0.03)
adam_lr = adam_lr0 · √(r/r0)            r = B·seq_len / T
epsilon = epsilon0 · √(r0/r)
beta2   = clip(beta2_0^(B/B0), 0.9, 0.9999)
beta1   = 0.668 (v0.6 reference override of heuristic default 0.9)
```

The optimizer hparams depend only on `(B, T)`, not on model size — so 1B and 4B get **identical** `lr / adam_lr / epsilon / beta2` for the same batch + token horizon.

For our setting (`B = 8192`, `T = 57.3×10⁹`), the per-run hparams resolve to approximately:
- `lr ≈ 4.49e-3`, `adam_lr ≈ 4.41e-3`, `beta2 ≈ 0.952`, `epsilon ≈ 6.77e-15`.

(Exact values come from `CompletedAdamHHeuristic.build_optimizer_config(...)` — verified with `PREVIEW_MODE=yes` before launch.)

`batch_size = 8192`, `per_device_parallelism = 1024` (matches exp135 1B settings; revisit per-device parallelism for 4B if it OOMs).

### LR schedule

Linear with **warmup = 0.1** (fraction of `num_train_steps`), **decay = 0.2** (cooldown phase to `min_lr_ratio = 0.0`). Defaults baked into `CompletedAdamHHeuristic`. Same schedule for 1B and 4B.

### Token horizon (1 epoch)

| Quantity | Value |
|---|---|
| Train samples | 223,880,280 |
| Sequence length | 256 (255 bp + BOS) |
| Total tokens (1 epoch) | 57,313,351,680 (~57.3B) |
| Batch size | 8,192 (matches exp135) |
| `num_train_steps` | ceil(223,880,280 / 8,192) = **27,329** |

Compare against #160's 0.6B baseline: 10,000 steps × 4,096 batch × 256 tokens = 10.5B tokens. This 1B/4B run consumes ~5.5× more tokens than #160.

### Eval / checkpoint cadence

- `EVALS_PER_RUN = 10` (matches exp135) → eval every `ceil(27,329 / 10) = 2,733` steps. Each eval covers TraitGym Mendelian v2 + LL gap on the four #160 validation regions.
- `CHECKPOINTS_PER_RUN = 3` (~every 9,110 steps) plus a 1-hour wall-clock checkpoint interval (whichever fires first), so checkpoints are at most ~1 hour apart even if the step-based interval is wider.

### Evaluations (match #160)

1. **TraitGym Mendelian v2** — full 9-subset suite (`distal`, `tss_proximal`, `missense`, `synonymous`, `splicing`, `non_coding_transcript_exon_variant`, `5_prime_UTR_variant`, `3_prime_UTR_variant`, `all`).
2. **LL gap on four region-specific validation sets** — `LL gap = loss(non-functional) − loss(functional)`, on each of:
   - [`bolinas-dna/genomes-v5-validation-intervals-v30_255_255`](https://huggingface.co/datasets/bolinas-dna/genomes-v5-validation-intervals-v30_255_255) — enhancers
   - [`bolinas-dna/genomes-v5-validation-intervals-v5_255_255`](https://huggingface.co/datasets/bolinas-dna/genomes-v5-validation-intervals-v5_255_255) — CDS
   - [`bolinas-dna/genomes-v5-validation-intervals-v1_255_255`](https://huggingface.co/datasets/bolinas-dna/genomes-v5-validation-intervals-v1_255_255) — upstream (promoter)
   - [`bolinas-dna/genomes-v5-validation-intervals-v15_255_255`](https://huggingface.co/datasets/bolinas-dna/genomes-v5-validation-intervals-v15_255_255) — downstream (3′-UTR)

   Same four val sets across both 1B and 4B runs to keep gap measurements apples-to-apples — and apples-to-apples with #160's 0.6B baseline. None of the four matches the zoonomia training distribution (those are conservation-filtered cross-mammal windows, not genomes-v5 annotation regions), so LL gap is a cross-domain signal, as in #160.

### Code

Marin branch (e.g. `eac/dna-bolinas-zoonomia-1ep`), module `experiments/dna/exp<this_issue_num>_zoonomia_1ep_scaling.py`. Adapt from [`exp135_bolinas_mix_sweep.py`](https://github.com/marin-community/marin/blob/eac/dna-bolinas-mix-sweep/experiments/dna/exp135_bolinas_mix_sweep.py): replace the multi-region `MIX_CONFIGS` with a single-component config pointing at `zoonomia-v1-v1`, sweep `MODEL_HIDDEN_SIZE` over the 1B and 4B values from `_build_model_configs`, set `TPU_TYPES` per model size, and remove the `MAX_EXAMPLES_PER_COMPONENT` epoch cap so the run consumes the full dataset.

Analysis scripts under this repo at `scripts/exp<this_issue_num>_zoonomia_1ep_scaling/` (collect from wandb only, per #109 pattern).

## TODO

- [ ] Confirm 4B fits on `v6e-4` and tune per-device parallelism (smoke-test before committing the full epoch — exp109 had `p4B` on `v6e-4` but at slow throughput; revisit if it doesn't fit the 27,329-step budget).
- [ ] Initial preview run (`PREVIEW_MODE=yes`) to verify token counts and per-run transferred hparams.
- [ ] Smoke test (`WARMUP_MODE=yes`, ~100 steps) for both sizes before launching the full epoch.

## Results

_Offline `evals_v2` matched-pair AUPRC for the exp166-v0.1 **1B** and **4B** runs, across the three HF-export checkpoints each (steps 10000 / 20000 / 27329). Pipeline: [`snakemake/analysis/evals_v2`](https://github.com/Open-Athena/marin-dna/tree/8c54ac44a37ecfc0ba8b6b1683b0ecbbb89ed118/snakemake/analysis/evals_v2) ([config](https://github.com/Open-Athena/marin-dna/blob/8c54ac44a37ecfc0ba8b6b1683b0ecbbb89ed118/snakemake/analysis/evals_v2/config/config.yaml)); datasets `bolinas-dna/evals_mendelian_traits` + `evals_complex_traits` (PR #194 k=9 rebuild) on GRCh38; metric = AUPRC ± paired cluster-bootstrap SE, `minus_llr_avg` (mendelian) / `abs_llr_avg` (complex), FWD+RC-averaged — the protocol the dashboard shows by default. Config + dashboard entries: #215. Both finals now appear on the mendelian + complex leaderboards._

> **Note — this is not the same metric as the WandB TraitGym AUPRC** in the [PR #5530 final-results comment](https://github.com/marin-community/marin/pull/5530#issuecomment-4583714149). That is an in-training eval on a different dataset construction; this is the offline matched-pair / cluster-bootstrap pipeline on the PR #194 k=9 datasets. Absolute values differ (offline mendelian global ≈ 0.27 vs WandB "all" ≈ 0.42); only the **1B-vs-4B comparison within these tables** is meaningful.

### Headline — final step 27329

| dataset | metric | 1B | 4B | Δ (4B−1B) |
|---|---|--:|--:|--:|
| mendelian | global AUPRC | 0.273 ± 0.009 | 0.297 ± 0.010 | +0.024 |
| mendelian | macro-avg AUPRC | 0.316 ± 0.014 | 0.339 ± 0.014 | +0.023 |
| complex | global AUPRC | 0.149 ± 0.006 | 0.151 ± 0.006 | +0.002 |
| complex | macro-avg AUPRC | 0.181 ± 0.015 | 0.196 ± 0.017 | +0.015 |

**4B beats 1B on mendelian** at every checkpoint (global + macro-avg). On **complex traits** the two are tied at the global level (within SE); 4B edges the macro-average.

### Mendelian convergence — global / macro AUPRC across the three exports

| step | 1B global | 4B global | 1B macro | 4B macro |
|---|--:|--:|--:|--:|
| 10000 | 0.258 ± 0.009 | 0.269 ± 0.009 | 0.276 ± 0.013 | 0.307 ± 0.014 |
| 20000 | 0.263 ± 0.009 | 0.286 ± 0.010 | 0.295 ± 0.014 | 0.317 ± 0.015 |
| 27329 | 0.273 ± 0.009 | 0.297 ± 0.010 | 0.316 ± 0.014 | 0.339 ± 0.014 |

Both sizes rise **monotonically** across the exported steps — the offline metric shows no late-training regression here. The HF exports (10k / 20k / 27329) do not bracket the WandB AUPRC peak at **step 24597**, so this coarse 3-point curve cannot resolve the in-training U-shape (the net 20k → 27329 change is positive for both sizes).

![exp166 mendelian AUPRC vs training step, 1B vs 4B, faceted by consequence subset](https://gist.githubusercontent.com/gonzalobenegas/3649e68fb63ca1f3443e4486078eb4d8/raw/318d2d63d87c88814c5e99c19ee56e7187df8c4e/exp166_zoonomia_scaling_auprc.svg)

_AUPRC vs training step (offline `minus_llr_avg`) for every mendelian subset plus the `_global_` / `_macro_avg_` aggregates; 1B (blue) vs 4B (red), error bars = paired cluster-bootstrap SE. Recipe: [`plots/exp166_zoonomia_scaling.py`](https://github.com/Open-Athena/marin-dna/blob/a3a7a3d99a7403a9369695f0fc18487cefc87d9c/plots/exp166_zoonomia_scaling.py). Complex omitted — evaluated at the final checkpoint only, so no trajectory._

### Per-subset — final step 27329

<details><summary><b>Mendelian</b>, per consequence subset (<code>minus_llr_avg</code> AUPRC ± SE)</summary>

| subset | 1B | 4B | Δ (4B−1B) | n_grp |
|---|--:|--:|--:|--:|
| `_global_` | 0.273 ± 0.009 | 0.297 ± 0.010 | +0.024 | 1614 |
| `_macro_avg_` | 0.316 ± 0.014 | 0.339 ± 0.014 | +0.023 | 8 |
| `3_prime_UTR_variant` | 0.193 ± 0.032 | 0.195 ± 0.028 | +0.002 | 77 |
| `5_prime_UTR_variant` | 0.280 ± 0.033 | 0.288 ± 0.032 | +0.007 | 210 |
| `distal` | 0.424 ± 0.061 | 0.405 ± 0.068 | -0.019 | 58 |
| `mature_miRNA_variant` | 0.521 ± 0.286 | 0.676 ± 0.233 | +0.155 | 4 |
| `missense_variant` | 0.284 ± 0.015 | 0.269 ± 0.014 | -0.015 | 580 |
| `non_coding_transcript_exon_variant` | 0.331 ± 0.038 | 0.397 ± 0.036 | +0.066 | 115 |
| `splicing` | 0.428 ± 0.025 | 0.468 ± 0.026 | +0.040 | 319 |
| `synonymous_variant` | 0.417 ± 0.063 | 0.404 ± 0.060 | -0.014 | 46 |
| `tss_proximal` | 0.167 ± 0.018 | 0.286 ± 0.030 | +0.119 | 205 |

</details>

<details><summary><b>Complex</b>, per consequence subset (<code>abs_llr_avg</code> AUPRC ± SE)</summary>

| subset | 1B | 4B | Δ (4B−1B) | n_grp |
|---|--:|--:|--:|--:|
| `_global_` | 0.149 ± 0.006 | 0.151 ± 0.006 | +0.002 | 1163 |
| `_macro_avg_` | 0.181 ± 0.015 | 0.196 ± 0.017 | +0.015 | 6 |
| `3_prime_UTR_variant` | 0.150 ± 0.042 | 0.155 ± 0.039 | +0.005 | 49 |
| `5_prime_UTR_variant` | 0.202 ± 0.060 | 0.240 ± 0.060 | +0.038 | 37 |
| `distal` | 0.162 ± 0.012 | 0.166 ± 0.012 | +0.005 | 616 |
| `missense_variant` | 0.165 ± 0.013 | 0.156 ± 0.013 | -0.009 | 250 |
| `non_coding_transcript_exon_variant` | 0.149 ± 0.037 | 0.223 ± 0.060 | +0.074 | 37 |
| `splicing` | 0.213 ± 0.069 | 0.236 ± 0.061 | +0.023 | 19 |
| `synonymous_variant` | 0.124 ± 0.024 | 0.108 ± 0.023 | -0.016 | 17 |
| `tss_proximal` | 0.257 ± 0.034 | 0.237 ± 0.034 | -0.021 | 138 |

</details>

**Where 4B helps:** gains concentrate in **regulatory / non-coding** subsets — mendelian `tss_proximal` (+0.119), `non_coding_transcript_exon` (+0.066), `splicing` (+0.040); complex `non_coding_transcript_exon` (+0.074), `5_prime_UTR` (+0.038). Coding subsets are flat-to-slightly-negative at 4B (mendelian `missense` −0.015, `synonymous` −0.014). (`mature_miRNA` shows a large 4B gain but only n=4 match-groups — not reliable.)

**Provenance:** metrics parquets at `s3://oa-bolinas/snakemake/analysis/evals_v2/results/metrics/exp166-v0.1-p{1B,4B}-step-{10000,20000,27329}/`; WandB group [`dna-bolinas-exp166-v0.1`](https://wandb.ai/gonzalobenegas/marin/groups/dna-bolinas-exp166-v0.1). Compute: 5× A10G (g5.xlarge, us-east-2), one short inference job per checkpoint, `--down` on idle.

## Links

- Research questions: #437
